src/Agent.Sdk/Util/StringUtil.cs

```diff
@@ -233,5 +233,19 @@ private static void EnsureLoaded()
                 s_locStrings = locStrings;
             }
         }
+
+        public static string HashNormalizer(string hash)
+        {
+            // Reducing the hash to the lower case without hyphens.
+            // For example: from "A1-B2-C3-D4-E5-F6", we get "a1b2c3d5f6".
+            return String.Join("", hash.Split("-")).ToLower();
+        }
+
+        public static bool HashesMatch(string hash1, string hash2)
```

Comment:
Nit: Since we are actually checking if hashes are equal to each other. I believe that we could call this method `AreHashesEqual`.
```suggestion
        public static bool AreHashesEqual(string leftValue, string rightValue)
```